projects/project6/SANSA-Template-Maven-Spark-develop/src/main/scala/evo/arules/mineRules.scala

```diff
@@ -0,0 +1,242 @@
+package evo.arules
+
+import ds._
+import evo.arules._
+
+import scala.util.Random
+import scala.collection.mutable.ListBuffer
+import java.net.URI
+
+import scala.collection.mutable
+import org.apache.jena.graph.NodeFactory
+import net.sansa_stack.rdf.spark.model._
+import net.sansa_stack.rdf.spark.io._
+import org.apache.spark.sql.SparkSession
+import org.apache.jena.riot.Lang
+
+import org.semanticweb.owlapi.model._
+import org.semanticweb.owlapi.model.OWLAxiom
+import net.sansa_stack.owl.spark.owl._
+import org.apache.spark.rdd.RDD
+
+// This class run the main algorithm and loads the ontology. It parses the owl file. Create a spark session
+// generated the required axioms.
+
+object mineRules extends App {
+
+
+  var a_names = ListBuffer[String]("A", "B", "C", "D", "E","F","G", "H","I","J","K","L","M","N","O","P")
+  var a_types = ListBuffer[String]("Role", "Concept")
+
+  var r = new Rule()
+  //println("Hello World"+r)
+
+  val ops = new Operations()
+  //create_dataset()
+
+
+  lazy val spark = SparkSession.builder().appName("MultiAssociation Rule Mining  $input").master("local[*]")
+    .config(
+      "spark.kryo.registrator",
+      "net.sansa_stack.owl.spark.dataset.UnmodifiableCollectionKryoRegistrator")
+    .getOrCreate()
+
+  type OWLAxiomsRDD = RDD[OWLAxiom]
```

Comment:
You do not need to define this type. It will be attached when the reader (`spark.owl...`) will happen (see my comment below).